lib/src/rules/lines_longer_than_80_chars.dart

```diff
@@ -0,0 +1,170 @@
+  LinesLongerThan80Chars()
+      : super(
+            name: 'lines_longer_than_80_chars',
+            description: _desc,
+            details: _details,
+            group: Group.style);
+
+  @override
+  void registerNodeProcessors(NodeLintRegistry registry) {
+    final visitor = new _Visitor(this);
+    registry.addCompilationUnit(this, visitor);
+  }
+}
+
+class _Visitor extends SimpleAstVisitor {
+  final LintRule rule;
+
+  _Visitor(this.rule);
+
+  @override
+  void visitCompilationUnit(CompilationUnit node) {
+    final lineInfo = node.lineInfo;
+
+    final allowedLineVisitor = new _AllowedLongLineVisitor(lineInfo);
+    node.accept(allowedLineVisitor);
+    final allowedCommentVisitor = new _AllowedCommentVisitor(lineInfo);
+    node.accept(allowedCommentVisitor);
+
+    final allowedLines = []
+      ..addAll(allowedLineVisitor.allowedLines)
+      ..addAll(allowedCommentVisitor.allowedLines);
+
+    final lineCount = lineInfo.lineCount;
+    for (int i = 0; i < lineCount; i++) {
+      final start = lineInfo.getOffsetOfLine(i);
+      final end = i == lineCount - 1
+          ? node.end
+          : lineInfo.getOffsetOfLineAfter(start) - 1;
+      final length = end - start;
+      if (length > 80 && !allowedLines.contains(i + 1)) {
+        rule.reporter.reportErrorForOffset(rule.lintCode, start, length);
+      }
+    }
+  }
+}
+
+class _AllowedLongLineVisitor extends RecursiveAstVisitor {
+  _AllowedLongLineVisitor(this.lineInfo);
+
+  final LineInfo lineInfo;
+  final allowedLines = <int>[];
+
+  @override
+  visitStringInterpolation(StringInterpolation node) {
+    if (node.isMultiline) {
+      _handleMultilines(node);
+    } else {
+      final value = node.elements.map((e) {
+        if (e is InterpolationString) return e.value;
+        if (e is InterpolationExpression) return ' ' * e.length;
+      }).join();
+      _handleSingleLine(node, value);
+    }
+  }
+
+  @override
+  visitSimpleStringLiteral(SimpleStringLiteral node) {
+    if (node.isMultiline)
+      _handleMultilines(node);
+    else
+      _handleSingleLine(node, node.value);
+  }
+
+  _handleMultilines(SingleStringLiteral node) {
+    final startLine = lineInfo.getLocation(node.offset).lineNumber;
+    final endLine = lineInfo.getLocation(node.end).lineNumber;
+    for (var i = startLine; i <= endLine; i++) {
+      allowedLines.add(i);
+    }
+  }
+
+  _handleSingleLine(AstNode node, String value) {
+    if (_looksLikeUriOrPath(value)) {
+      final line = lineInfo.getLocation(node.offset).lineNumber;
+      allowedLines.add(line);
+    }
+  }
+}
+
+class _AllowedCommentVisitor extends GeneralizingAstVisitor {
+  _AllowedCommentVisitor(this.lineInfo);
+
+  final LineInfo lineInfo;
+  final allowedLines = <int>[];
+
+  @override
+  visitNode(AstNode node) {
+    final comments = <CommentToken>[]
+      ..addAll(_getPrecedingComments(node.beginToken))
+      ..addAll(_getPrecedingComments(node.endToken));
```

Comment:
This skips comments preceding either the first or last token. It also examines many tokens multiple times. It's probably better to ask the compilation unit for the first token and for each token ask for preceding comments.